doc/v5.0/man3/MPI_Get_count.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_Get_count" "3" "" "2021-08-30" "Open MPI"
.hy
.SH Name
.PP
\f[C]MPI_Get_count\f[R] - Gets the number of top-level elements
received.
.SH Syntax
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>

int MPI_Get_count(const MPI_Status *status, MPI_Datatype datatype,
    int *count)
\f[R]
.fi
.SS Fortran Syntax
.IP
.nf
\f[C]
USE MPI
! or the older form: INCLUDE \[aq]mpif.h\[aq]

MPI_GET_COUNT(STATUS, DATATYPE, COUNT, IERROR)
    INTEGER STATUS(MPI_STATUS_SIZE), DATATYPE, COUNT, IERROR
\f[R]
.fi
.SS Fortran 2008 Syntax
.IP
.nf
\f[C]
USE mpi_f08

MPI_Get_count(status, datatype, count, ierror)
    TYPE(MPI_Status), INTENT(IN) :: status
    TYPE(MPI_Datatype), INTENT(IN) :: datatype
    INTEGER, INTENT(OUT) :: count
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
\f[R]
.fi
.SH Input Parameters
.IP \[bu] 2
\f[C]status\f[R] : Return status of receive operation (status).
.IP \[bu] 2
\f[C]datatype\f[R] : Datatype of each receive buffer element (handle).
.SH Output Parameters
.IP \[bu] 2
\f[C]count\f[R] : Number of received elements (integer).
.IP \[bu] 2
\f[C]IERROR\f[R] : Fortran only: Error status (integer).
.SH Description
.PP
Returns the number of entries received.
(We count entries, each of type \f[C]datatype\f[R], not bytes.) The
\f[C]datatype\f[R] argument should match the argument provided by the
receive call that set the \f[C]status\f[R] variable.
(As explained in Section 3.12.5 in the MPI-1 Standard, \[lq]Use of
General Datatypes in Communication,\[rq] \f[C]MPI_Get_count\f[R] may, in
certain situations, return the value \f[C]MPI_UNDEFINED\f[R].)
.PP
The \f[C]datatype\f[R] argument is passed to \f[C]MPI_Get_count\f[R] to
improve performance.
A message might be received without counting the number of elements it
contains, and the \f[C]count\f[R] value is often not needed.
Also, this allows the same function to be used after a call to
\f[C]MPI_Probe\f[R].
.SH Notes
.PP
If the size of the \f[C]datatype\f[R] is zero, this routine will return
a \f[C]count\f[R] of zero.
If the amount of data in \f[C]status\f[R] is not an exact multiple of
the size of \f[C]datatype\f[R] (so that \f[C]count\f[R] would not be
integral), a \f[C]count\f[R] of \f[C]MPI_UNDEFINED\f[R] is returned
instead.
.SH Errors
.PP
If the value to be returned is larger than can fit into the
\f[C]count\f[R] parameter, an \f[C]MPI_ERR_TRUNCATE\f[R] error is
raised.
.PP
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
.PP
Before the error value is returned, the current MPI error handler is
called.
By default, this error handler aborts the MPI job, except for I/O
function errors.
The error handler may be changed with \f[C]MPI_Comm_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_RETURN\f[R] may be used to
cause error values to be returned.
Note that MPI does not guarantee that an MPI program can continue past
an error.
.SH See Also
.PP
\f[C]MPI_Get_elements\f[R](3)


doc/v1.4/man3/MPI_Get_count.3.php
<?php
$topdir = "../../..";
$title = "MPI_Get_count(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_GET_COUNT(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Get_count </b> - Gets the number of top-level elements received.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C
Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Get_count(MPI_Status *status, MPI_Datatype datatype,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;int *count)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_GET_COUNT(STATUS, DATATYPE, COUNT, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;STATUS(MPI_STATUS_SIZE), DATATYPE, COUNT, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int Status::Get_count(const Datatype&amp; datatype) const
</pre>
<h2><a name='sect5' href='#toc5'>Input Parameters</a></h2>

<dl>

<dt>status </dt>
<dd>Return status of receive operation (status). </dd>

<dt>datatype
</dt>
<dd>Datatype of each receive buffer element (handle).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameters</a></h2>

<dl>

<dt>count
</dt>
<dd>Number of received elements (integer). </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error status
(integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
Returns the number of entries received. (We count
entries, each of type datatype, not bytes.) The datatype argument should
match the argument provided by the receive call that set the status variable.
(As explained in Section 3.12.5 in the MPI-1 Standard, "Use of General Datatypes
in Communication," MPI_Get_count may, in certain situations, return the
value MPI_UNDEFINED.)   <p>
The datatype argument is passed to MPI_Get_count
to improve performance. A message might be received without counting the
number of elements it contains, and the count value is often not needed.
Also, this allows the same function to be used after a call to <a href="../man3/MPI_Probe.3.php">MPI_Probe</a>.

<p>
<h2><a name='sect8' href='#toc8'>Notes</a></h2>
If the size of the datatype is zero, this routine will return a count
of zero.  If the amount of data in  <i>status</i> is not an exact multiple of the
size of  <i>datatype</i> (so that  <i>count</i> would not be integral), a  <i>count</i> of <i>MPI_UNDEFINED</i>
is returned instead.
<p>
<h2><a name='sect9' href='#toc9'>Errors</a></h2>
Almost all MPI routines return an error value;
C routines as the value of the function and Fortran routines in the last
argument. C++ functions do not return errors. If the default error handler
is set to MPI::ERRORS_THROW_EXCEPTIONS, then on error the C++ exception
mechanism will be used to throw an MPI:Exception object. <p>
Before the error
value is returned, the current MPI error handler is called. By default,
this error handler aborts the MPI job, except for I/O function errors. The
error handler may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the predefined
error handler MPI_ERRORS_RETURN may be used to cause error values to be
returned. Note that MPI does not guarantee that an MPI program can continue
past an error.
<p>
<h2><a name='sect10' href='#toc10'>See Also</a></h2>
<p>
<a href="../man3/MPI_Get_elements.3.php">MPI_Get_elements</a>
<p>
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>C++ Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input Parameters</a></li>
<li><a name='toc6' href='#sect6'>Output Parameters</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Notes</a></li>
<li><a name='toc9' href='#sect9'>Errors</a></li>
<li><a name='toc10' href='#sect10'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
